### PR TITLE
feat(mongo): relative date support in filter step (mongo 5+) TCTC-1197

### PR DESCRIPTION
**Pull request by davinov**
Introduce a new `mongo50` translator, which knows how to translate `RelativeDate`s.

An important point: the format 
```{
  $match: {
    column: {
      $operator: value
    }
  }
}
```
does not work with `$dateAdd` expressions, so we had to use [`$expr`](https://docs.mongodb.com/manual/reference/operator/query/expr/).

Other note: in the `RelativeDate` type, there is no reference date, so I guessed this was going to be the current moment (`$$NOW`)

### CODE REVIEW

**Comment by davinov** [comment]
Still needs some work, because if we replace all filters by $expr, they will be performance implications:
- indexes will no be used for mongo before 5
- and even in mongo 5 there is some limitations

**Comment by raphaelvignes** [comment]
> indexes will no be used for mongo before 5
and even in mongo 5 there is some limitations 
:(

**Comment by piotch** [comment]
@davinov cool!

I had a question which is beyond the scope of the PR : when you said it is relative to `$$NOW` I guess it is the UTC value, is it obvious  because we make sure all dates passing thru these filters are UTC as well ? I am not sure where we stand on this. Even if it is obvious maybe it would be worth documenting in the relevant confluence page that it is how it works.

File: src/lib/translators/mongo.ts [hunk 1]
```diff
@@ -2261,26 +2261,26 @@ export class Mongo36Translator extends BaseTranslator {
   /**
    * Translate any value expression into one mongo can evaluate.
    *
-   * @private
+   * @protected
    */
-  private translateValue(value: any): any {
+  protected translateValue(value: any): any {
```
Comment (alice-sevin) [inline src/lib/translators/mongo.ts]: pourquoi on passe a du protected plutot qu'à du private ici ?

**Review by alice-sevin — APPROVED** [review]
LGTM! A prévoir aujourd'hui on renvoit toujours une quantity négative pour une relative date je crois. Il faudra penser a corriger ça car j'imagine que fonctionnelement pour l'instant le $add ne marchera pas comme espéré

**Pull request merged**